### ISSUE TITLE
[MJAVADOC-227] Aggregated javadocs do not take per project configuration into account

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#931]
**[Carsten Ziegeler](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cziegeler)** opened **[MJAVADOC-227](https://issues.apache.org/jira/browse/MJAVADOC-227?redirect=false)** and commented

We have a multi project setup where each module has a configuration for the javadocs for excluding packages.
Building the javadocs for each project separately works fine.
Starting the build from the root and building aggregated javadocs there, results in javadocs for all sources. The exclude package configurations of the individual poms are not considered.
I've used "javadoc:javadoc" to build the docs and tried both, putting the config in the plugin section and the reporting section in each module.

A workaround is of course to manually configure the javadoc plugin in the root pom and copy all excludes from the modules there; but that's a maintenance nightmare.

---

**Affects:** 2.5

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

IIUC you want to run javadoc:aggregate from root and use the subproject plugin configuration: it is not possible (maven core) and I close it.

I suggest you to add properties to exclude your packages in the root and use them in the sub projects.

**Comment by jira-importer** [comment]
**[Brian E. Fox (imported)](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brianf@infinity.nu)** commented

I'll add further clarification: The javadoc aggregate goal is, surprisingly, an aggregator. This means unlike normal goals that cause maven to iterate over each project, this one runs just once. It asks maven ahead of time to resolve all the projects in the reactor and it gets a list of them injected. The plugin configuration however comes from the project where it's being executed. Therefore, the configuration you have in all the sub modules is completely irrelevant as the only thing in context for configuration is the root project. If you want it to exclude certain things, you need to configure it here in the root or the plugin won't see it at all.

**Comment by jira-importer** [comment]
**[Carsten Ziegeler](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cziegeler)** commented

Yes, we figured this out and it caused a lot of pain as this requires to add all configs from all sub modules to the parent pom as well. Imagine a project with > 100 modules all with different excludes - it's a nightmare :) That's why I filed the bug and the other reason is that the current behaviour wasn't the behaviour I expected.
I understand that there might be technical reasons, however, if the root project already knows all its subprojects (in order to aggregate), why can't it just aggregate the configs of the children first?
